src/test/java/org/springframework/data/elasticsearch/JUnit5SampleRestClientBasedTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch;
+
+import static org.assertj.core.api.Assertions.*;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate;
+import org.springframework.data.elasticsearch.junit.jupiter.ClusterConnectionInfo;
+import org.springframework.data.elasticsearch.junit.jupiter.ElasticsearchRestTemplateConfiguration;
+import org.springframework.data.elasticsearch.junit.jupiter.SpringDataElasticsearch;
+import org.springframework.data.elasticsearch.junit.jupiter.SpringDataElasticsearchTest;
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * class demonstrating the setup of a JUnit 5 test in Spring Data Elasticsearch that uses the rest client. The
+ * ContextConfiguration must include the test class itself (and provide a
+ * {@link org.springframework.data.elasticsearch.junit.jupiter.ClusterConnectionInfo} bean) and the
+ * {@link ElasticsearchRestTemplateConfiguration} class
+ *
+ * @author Peter-Josef Meisch
+ */
+@SpringDataElasticsearchTest
+@ContextConfiguration(
+		classes = { JUnit5SampleRestClientBasedTests.class, ElasticsearchRestTemplateConfiguration.class })
```

Comment:
It's not a good idea to use a test class as a pseudo `@Configuration` class (even it "lite bean mode").

Rather, you should use a static nested `@Configuration` class within the test class for such purposes, and that class can then `@Import` additional configuration like the `ElasticsearchRestTemplateConfiguration`.